File: JavaPractice/src/ExtractDigitsFromStringToSum.java

import java.util.regex.*;

public class ExtractDigitsFromStringToSum {

	public static void ExtractDigitsToSum(String s) {
		Pattern p = Pattern.compile("\\-?\\d+"); // - is to consider -(-ve) and ? is to check option if it is there or
		// not,
		// \\d is to consider any digit from 0-9 and then + will be for more
		// then one
		// another way to add can be "\\-?[0-9]+
		Matcher m = p.matcher(s);
		int sum = 0;
		int n = 0;
		while (m.find()) {
			n = Integer.parseInt(m.group());
			sum += n;
		}
		System.out.println(sum);

	}

	static int addSummation(String inputString) {

		if (inputString == null || inputString.length() < 2)
			throw new IllegalArgumentException("Pass valid input");

		char[] inputChar = inputString.toCharArray();
		boolean flag = true;
		int sum = 0;
		String result = "";

		for (int i = 0; i < inputChar.length; i++) {

			if (Character.isDigit(inputChar[i])) {
				result = result + inputChar[i];
				flag = true;
				//System.out.println(result);
			} else {
				try { 
					if(!Character.isDigit(inputChar[i]) && flag) {
						sum += Integer.parseInt(result);
						flag = false;
						result = "";
					}
				}
				catch (NumberFormatException e) {
					result = "";
				}
			}

		}
		if(!result.isEmpty())
			sum += Integer.parseInt(result);
		return sum;

	}

	public static void main(String[] args) {
		// TODO Auto-generated method stub

		System.out.println("*** Extract Digits Method ***");
		ExtractDigitsToSum("23adf5dsf4"); 
		ExtractDigitsToSum("235a4");
		ExtractDigitsToSum("2354a"); 
		ExtractDigitsToSum("a2354");
		ExtractDigitsToSum("a2a3a5a4a"); 
		ExtractDigitsToSum("123a-bc-22");

		System.out.println("\n*** ******** ***\n");
		System.out.println("*** Add summation method ***");
		System.out.println(addSummation("123abc45def6gh78"));
		System.out.println(addSummation("23adf5dsf4"));
		System.out.println(addSummation("235a4"));
		System.out.println(addSummation("2354a"));
		System.out.println(addSummation("a2a3a5a4a"));
		System.out.println(addSummation("232345"));
		System.out.println(addSummation("abcde"));
		System.out.println(addSummation("1abcde1"));
	}

}

File: PythonPractice/src/ExtractDigitsFromStringToSum.py
import re
from typing import Optional

def extract_digits_to_sum(s: str) -> None:
    if not isinstance(s, str):
        raise TypeError("Input must be a string")

    pattern = re.compile(r'-?\d+')
    matches = pattern.findall(s)

    sum_digits = 0
    for match in matches:
        try:
            sum_digits += int(match)
        except ValueError:
            # If there's an issue with converting to int, skip this match
            continue

    print(sum_digits)

def add_summation(input_string: Optional[str]) -> int:
    if not isinstance(input_string, str):
        raise TypeError("Input must be a string")
    if len(input_string) < 2:
        raise ValueError("Input string is too short")

    sum_digits = 0
    result = ''
    flag = False

    for char in input_string:
        if char.isdigit():
            result += char
            flag = True
        else:
            if flag:
                try:
                    sum_digits += int(result)
                except ValueError:
                    # If there's an issue with converting to int, reset result
                    result = ''
                result = ''
                flag = False

    if result:
        try:
            sum_digits += int(result)
        except ValueError:
            # If there's an issue with converting to int, ignore
            pass

    return sum_digits

# Example usage
print("*** Extract Digits Method ***")
extract_digits_to_sum("23adf5dsf4")
extract_digits_to_sum("235a4")
extract_digits_to_sum("2354a")
extract_digits_to_sum("a2354")
extract_digits_to_sum("a2a3a5a4a")
extract_digits_to_sum("123a-bc-22")

print("\n*** ******** ***\n")
print("*** Add Summation Method ***")
print(add_summation("123abc45def6gh78"))
print(add_summation("23adf5dsf4"))
print(add_summation("235a4"))
print(add_summation("2354a"))
print(add_summation("a2a3a5a4a"))
print(add_summation("232345"))
print(add_summation("abcde"))
print(add_summation("1abcde1"))
